### PR TITLE
PX4 NuttX kernel mode build part 0.1

### PR DESCRIPTION
**Pull request by pussuw**
Add a skeleton to enable building PX4 as a set of processes on NuttX with CONFIG_BUILD_KERNEL=y

The result is a bootable ROMFS which is placed into the kernel binary and mounted into /sbin by the board initialization logic. The kernel knows the mount point and starts the user space from /sbin/init, which is the nsh shell.

### CODE REVIEW

**Comment by pussuw** [comment]
DO NOT MERGE; ONLY INTENDED TO HOLD THE PROGRESS IN A SAFE PLACE AND FOR COMMENTS.

File: platforms/nuttx/CMakeLists.txt [hunk 1]
```diff
@@ -241,8 +257,71 @@ if (NOT CONFIG_BUILD_FLAT)
 
 		DEPENDS px4 px4_kernel
 	)
-
 else()
+## TODO REMOVE ME
```
Comment (jlaitine) [inline platforms/nuttx/CMakeLists.txt:261]: ?

File: platforms/nuttx/CMakeLists.txt [hunk 2]
```diff
@@ -241,8 +257,71 @@ if (NOT CONFIG_BUILD_FLAT)
...
+
+
+		m
+		gcc
+		)
+
+	set(nuttx_userspace)
+
+	list(APPEND nuttx_userspace
+		#drivers_userspace # init_entrypt.c
```
Comment (jlaitine) [inline platforms/nuttx/CMakeLists.txt:285]: Is this waiting for the next step, i.e. temporarily commented out?

**Review by jlaitine — APPROVED** [review]
LGTM!

File: platforms/nuttx/CMakeLists.txt (hunk 1 above)
Comment (pussuw) [reply platforms/nuttx/CMakeLists.txt:261]: This PR needs clean up, I posted it just to a) keep the changes somewhere safe and b) to get initial comments about the changes. Obviously I will address all of the TODOs and quirks.

File: platforms/nuttx/CMakeLists.txt (hunk 2 above)
Comment (pussuw) [reply platforms/nuttx/CMakeLists.txt:285]: Yes need to modify this list. Building nsh does not require this lib, but it does not matter if it is still in this list.

**Comment by pussuw** [comment]
Closing as obsolete

**Pull request closed without merging**